microceph/ceph/operations.go

```diff
@@ -0,0 +1,262 @@
+package ceph
+
+import (
+	"fmt"
+
+	"github.com/canonical/lxd/shared/logger"
+
+	microCli "github.com/canonical/microcluster/v2/client"
+
+	"github.com/canonical/microceph/microceph/client"
+)
+
+// RunOperations runs the provided operations or prints out the action plan.
+func RunOperations(name string, operations []Operation, dryRun bool) error {
+	for _, ops := range operations {
+		if dryRun {
+			fmt.Println(ops.DryRun(name))
+		} else {
+			err := ops.Run(name)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Operation is a interface for ceph and microceph operations.
+//
+type Operation interface {
+	// Run executes the operation and return the error if any.
+	Run(string) error
+
+	// DryRun returns the string representation of the operation.
+	DryRun(string) string
+}
+
+// CheckNodeInClusterOps is an operation to check if a node is in the microceph cluster.
+type CheckNodeInClusterOps struct {
+	CephClient    client.ClientInterface
+	ClusterClient *microCli.Client
+}
```

Comment:
Suggestion: introduce a base class `ClusterOps` with the CephClient and ClusterClient fields, then all the other classes / structs could inherit from `ClusterOps`